// -*- mode: c++; indent-tabs-mode: nil; -*-
//
// Copyright (c) 2017 Illumina, Inc.
// All rights reserved.

// Redistribution and use in source and binary forms, with or without
// modification, are permitted provided that the following conditions are met:

// 1. Redistributions of source code must retain the above copyright notice, this
//    list of conditions and the following disclaimer.

// 2. Redistributions in binary form must reproduce the above copyright notice,
//    this list of conditions and the following disclaimer in the documentation
//    and/or other materials provided with the distribution.

// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
// ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
// WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
// OR TORT INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

syntax = "proto3";

package reads;

enum MappingStatus {
    UNMAPPED = 0;
    MAPPED = 1;
    BAD_ALIGN = 2;
}

message Read {
    string fragment_id = 1;
    string bases = 2;
    string quals = 3;
    int32 chrom_id = 4;
    int32 pos = 5;
    uint32 mapq = 6;

    bool is_reverse_strand = 10;
    bool is_mate_reverse_strand = 11;
    bool is_mapped = 12;
    bool is_first_mate = 13;
    bool is_mate_mapped = 14;
    int32 mate_chrom_id = 15;
    int32 mate_pos = 16;

    int32 graph_pos = 100;
    string graph_cigar = 101;
    int32 graph_mapq = 102;
    int32 graph_alignment_score = 103;
    bool is_graph_alignment_unique = 104;
    bool is_graph_reverse_strand = 105;

    repeated string graph_nodes_supported = 106;
    repeated string graph_edges_supported = 107;
    repeated string graph_sequences_supported = 108;
    repeated string graph_sequences_broken = 109;

    MappingStatus graph_mapping_status = 200;
}

